### PR TITLE
Appointment Notifications

### PR DESCRIPTION
**Pull request by lionsdwarf**
Appointment notifications for: 
1) Individual Appointments and Group Attendees confirmed within the last hour 
2) Individual Appointments and Group Attendees with Appointment `starts_at` within the upcoming 24 hours.

Two things currently needed to make this prime-time ready:
1) Timezone datetime coercion 
2) Worker config

### CODE REVIEW

File: apps/api/services/notifications.js [hunk 1]
```diff
@@ -1,17 +1,51 @@
...
+    })
+  )
+  apptRollups.content = content
+  return apptRollups
+}
+
+export const sendAppointmentNotification = apptRollups =>
+  Promise.all(
+    apptRollups.map(appt => {
+      return sendEmail(appt.recipient, apptRollups.content, {
```
Comment (craigspaeth) [inline apps/api/services/notifications.js]: Nit-pick: You can leave out the `return` and curlies here to win at code ⛳️

File: apps/api/services/notifications.js [hunk 2]
```diff
@@ -1,17 +1,51 @@
...
-    filter: `name = 'assignedSalve'`
+    filter: `name = '${name}'`
   }))[0]
+
+export const upcomingAppointments = async () =>
+  appointmentRollups(await Appointment.findStartsBetween(), 'remindAppointment')
+
+export const confirmedAppointments = async () =>
+  appointmentRollups(
+    await Appointment.findCreatedWithin(),
```
Comment (craigspaeth) [inline apps/api/services/notifications.js]: Nit-pick: The defaults actually make this kind hard to read—might be better to leave it exposed 🤷‍♂

**Comment by craigspaeth** [comment]
👏 really good stuff here! This is an excellent starting point. Also one more item for the prime-time list: tests.

File: apps/api/services/notifications.js (hunk 1 above)
Comment (lionsdwarf) [reply apps/api/services/notifications.js]: Good catch. Thxxx

File: apps/api/services/notifications.js (hunk 2 above)
Comment (lionsdwarf) [reply apps/api/services/notifications.js]: What do you mean by exposed exactly?

**Comment by lionsdwarf** [comment]
Closing this for now. Thanks for the feedback. Will definitely add tests.

File: apps/api/services/notifications.js [hunk 3]
```diff
@@ -1,17 +1,51 @@
...
+    'confirmAppointment'
+  )
+
+export const appointmentRollups = async (appts, contentName) => {
+  const content = await findContentByName(contentName)
+  const apptRollups = await Promise.all(
+    appts.map(async appt => {
+      appt.recipient = await Patient.findForAppointment(appt)
```
Comment (craigspaeth) [inline apps/api/services/notifications.js]: Noticed an issue here. This assumes a 1:1 patients to appointment—meaning it won't work for groups where there are many:1.

File: apps/api/services/notifications.js (hunk 2 above)
Comment (craigspaeth) [reply apps/api/services/notifications.js]: Just removing the defaults so it reads `Appointment.findCreatedWithin(moment(), moment().add(24, 'hours')` which reads a bit more explicitly to me. (like I said—just a nit pick)

**Pull request closed without merging**